### PR TITLE
[stdlib] Add docs for Sequence._customContainsEquatableElement

### PR DESCRIPTION
**Pull request by lorentey**
This underscored customization point is currently lacking documentation. Add some.

### CODE REVIEW

**Comment by lorentey** [comment]
@swift-ci smoke test

**Comment by lorentey** [comment]
@swift-ci smoke test windows platform

**Comment by lorentey** [comment]
@swift-ci test windows platform

File: stdlib/public/core/Sequence.swift [hunk 1]
```diff
@@ -355,6 +355,31 @@ public protocol Sequence<Element> {
...
+  /// This method must not return non-nil on sequences that are destructively
+  /// consumed by iteration.
+  ///
+  /// - Returns: `nil` if containment cannot be verified in better than linear
+  ///    time; otherwise, the method returns a boolean value indicating whether
+  ///    or not the item is an element of this sequence.
+  ///
+  /// - Complexity: If this function returns `nil`, it must do so in constant
+  ///    time. If this returns non-`nil`, then it must have better than linear
+  ///    complexity.
```
Comment (glessard) [inline stdlib/public/core/Sequence.swift:382]:
```suggestion
  ///    (O(1)) time. If this returns non-`nil`, then it must have better than linear
  ///    (O(*n*)) complexity.
```
Most "Complexity" documentation clauses uses O-notation, so while I appreciate the additional detail I'm not sure we should omit that notation entirely. Make of that what you will…

**Review by glessard — APPROVED** [review]
Good addition. I would add O-notation to the complexity clause for consistency with the rest of the documentation.

File: stdlib/public/core/Sequence.swift [hunk 2]
```diff
@@ -355,6 +355,31 @@ public protocol Sequence<Element> {
...
+  /// `Sequence` and `Collection` algorithms other than `contains` itself may
+  /// adapt their behavior based on whether or not this function returns nil.
+  /// For example, a generic algorithm that needs to do containment checks for
+  /// many different values may decide not to copy items into a temporary `Set`
+  /// if it sees that the sequence implements this method. Therefore, sequences
+  /// should only implement this method if they can do it in better than linear
+  /// time.
+  ///
+  /// This method must not return non-nil on sequences that are destructively
+  /// consumed by iteration.
```
Comment (natecook1000) [inline stdlib/public/core/Sequence.swift:374]: Is this a safe assumption/requirement to make? It seems like it should be, but this hasn't been communicated before. (Trying to think of a consuming sequence that could implement this in sub-linear time and not having any luck.)

File: stdlib/public/core/Sequence.swift (hunk 1 above)
Comment (lorentey) [reply stdlib/public/core/Sequence.swift:382]: We are generally careful not to say O(*n*) without defining what *n* is. However, this ultimately doesn't matter 👍

File: stdlib/public/core/Sequence.swift (hunk 2 above)
Comment (lorentey) [reply stdlib/public/core/Sequence.swift:374]: Maybe we should instead say that this method does not consume any elements:

```suggestion
  /// For sequences that are destructively consumed by iteration, calling this
  /// method must not consume any elements. (Such sequences usually leave this
  /// method with its default, `nil`-returning implementation, which trivially
  /// satisfies this requirement.)
```

**Comment by lorentey** [comment]
@swift-ci test

**Pull request merged**